axiom/nr_txn.c

```diff
@@ -157,6 +173,32 @@ void nr_txn_set_long_attribute(nrtxn_t* txn,
                                attribute->name, value);
 }
 
+void nr_txn_attributes_set_string_attribute(nr_attributes_t* attributes,
```

Comment:
Could this be tested with a unit test to verify it handles bad inputs, etc?

Reply:
Please see: [921224e](https://github.com/newrelic/newrelic-php-agent/pull/585/commits/921224e3c10ba59cc9249787ad25d1dcdf519d2a)